src/mscorlib/src/System/Lazy.cs

```diff
@@ -254,7 +456,7 @@ internal T ValueForDebugDisplay
                 {
                     return default(T);
                 }
-                return ((Boxed)m_boxed).m_value;
+                return Value;
```

Comment:
When using locking, if another thread is currently initializing the value while holding the lock, this will end up blocking, right?  I believe ValueForDebugDisplay was explicit coded to avoid the lock because the debugger evaluation could hang if it tries to take the lock that another thread (frozen by the debugger) is currently holding.

(I thought we were using NotifyOfCrossThreadDependency in Value, but it looks like that was removed for CoreCLR: https://referencesource.microsoft.com/#mscorlib/system/Lazy.cs,337. We should probably put that back to avoid this kind of issue; then your change here would be ok.)